app/root.tsx

```diff
@@ -63,46 +86,82 @@ export const Head = createHead(() => (
   </>
 ));
 
-export function Layout({ children }: { children: React.ReactNode }) {
+function ClientOnly({ children }: { children: React.ReactNode }) {
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => setMounted(true), []);
+
+  return mounted ? <>{children}</> : null;
+}
+
+function ThemeProvider() {
   const theme = useStore(themeStore);
 
   useEffect(() => {
     document.querySelector('html')?.setAttribute('data-theme', theme);
   }, [theme]);
 
-  return (
-    <>
-      {children}
-      <ScrollRestoration />
-      <Scripts />
-    </>
-  );
+  return null;
 }
 
-import { logStore } from './lib/stores/logs';
+function AuthProvider({ data }: { data: LoaderData }) {
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      window.ENV = data.ENV;
+      initializeAuth().catch((err: Error) => {
+        logStore.logError('Failed to initialize auth', err);
+      });
+    }
+  }, [data]);
+
+  return null;
+}
 
 export const ErrorBoundary = () => {
   const error = useRouteError();
-  sentryHandleError(error as Error);
+
+  // Using our conditional error handling instead of direct Sentry import
+  sentryHandleError(error instanceof Error ? error : new Error(String(error)));
 
   return <div>Something went wrong</div>;
 };
 
 export default function App() {
-  const theme = useStore(themeStore);
+  const data = useLoaderData<typeof loader>() as LoaderData;
+  const [mounted, setMounted] = useState(false);
 
   useEffect(() => {
-    logStore.logSystem('Application initialized', {
-      theme,
-      platform: navigator.platform,
-      userAgent: navigator.userAgent,
-      timestamp: new Date().toISOString(),
-    });
+    window.ENV = data.ENV;
+    setMounted(true);
   }, []);
 
+  // Only access stores on the client side
+  const theme = useStore(themeStore)
+  const user = useStore(userStore)
+  const isLoading = useStore(isLoadingStore)
+
+  useEffect(() => {
+    if (mounted) {
+      logStore.logSystem('Application initialized', {
+        theme,
+        platform: navigator.platform,
+        userAgent: navigator.userAgent,
+        timestamp: new Date().toISOString(),
+        isAuthenticated: !!user,
+      });
+    }
+  }, [theme, user, mounted]);
+
   return (
-    <Layout>
-      <Outlet />
-    </Layout>
+    <>
+      <ClientOnly>
```

Comment:
Are we disabling SSR for most of the application?
I don't know what the remifications of that are. Might need to tag in @Andarist for help 🙏

Reply:
I think this will make more of the app SPA like, but that should be fine because you wont do anything until it's interactive.